uber/models.py

```diff
@@ -1243,9 +1196,10 @@ def _badge_adjustments(self):
 
         self.badge_type = get_real_badge_type(self.badge_type)
 
+        if self.orig_value_of('badge_type') != self.badge_type:
+            self.session.update_badge(self, self.orig_value_of('badge_type'), self.orig_value_of('badge_num'))
```

Comment:
Yup, this version of the code didn't actually fix it. That's what I get for forgetting to add a regression test.